### PR TITLE
Bump version for System.IO

### PR DESCRIPTION
**Pull request by JeremyKuhne**
@weshaggard @ericstj @danmosemsft @joperezr

### CODE REVIEW

File: src/System.IO/src/System.IO.csproj [hunk 1]
```diff
@@ -4,10 +4,12 @@
   <PropertyGroup>
     <AssemblyName>System.IO</AssemblyName>
     <ProjectGuid>{07390899-C8F6-4e83-A3A9-6867B8CB46A0}</ProjectGuid>
-    <AssemblyVersion>4.1.0.0</AssemblyVersion>
+    <AssemblyVersion>4.2.0.0</AssemblyVersion>
+    <AssemblyVersion Condition="'$(TargetGroup)' == 'net462'">4.1.0.0</AssemblyVersion>
+	<ContractProject Condition="'$(TargetGroup)' == 'net462'">..\ref\4.1\System.IO.depproj</ContractProject>
```
Comment (weshaggard) [inline src/System.IO/src/System.IO.csproj]: nit: indention.

File: src/System.IO/ref/4.1/project.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "dependencies": {
+    "System.Runtime": "4.0.20",
```
Comment (weshaggard) [inline src/System.IO/ref/4.1/project.json]: This needs to be only "System.IO" "4.1.0" so bin-placing works as expected. This is why the CI legs are failing.

Comment (JeremyKuhne) [inline src/System.IO/ref/4.1/project.json]: That should be the only dependency?

File: src/System.IO/src/System.IO.csproj (hunk 1 above)
Comment (JeremyKuhne) [inline src/System.IO/src/System.IO.csproj]: Fixed spacing issues

File: src/System.IO/ref/4.1/project.json (hunk 2 above)
Comment (weshaggard) [inline src/System.IO/ref/4.1/project.json]: Yes similar to the other depproj's in the other version folders.

Comment (JeremyKuhne) [inline src/System.IO/ref/4.1/project.json]: Got it

**Comment by ericstj** [comment]
Per our discussion on Friday this is OK for now but it will not work on the 1.0 shared framework since that does not pull in the new implementation.

We should have an issue filed to move this to netstandard1.7/netcoreapp1.1 once that is available.

If we really needed to ship this to shared framework 1.0 and can do so against the old mscorlib/system.private.corelib an alternative would be to transition this to a fat package and keep it at netstandard1.6/netcoreapp1.0.

**Comment by JeremyKuhne** [comment]
@ericstj Do we have an overall issue for 1.7/1.1 that I can refer to?

**Comment by ericstj** [comment]
@weshaggard opened https://github.com/NuGet/Home/issues/2998 on NuGet.

**Comment by weshaggard** [comment]
I just filed https://github.com/dotnet/corefx/issues/9556 to track the general corefx moving to NS1.7

**Comment by JeremyKuhne** [comment]
test Innerloop OSX Debug Build and Test

**Comment by JeremyKuhne** [comment]
test Innerloop Windows_NT Release Build and Test

**Pull request merged**